pmd-core/src/test/java/net/sourceforge/pmd/cache/FileAnalysisCacheTest.java

```diff
@@ -130,6 +134,12 @@ void testStorePersistsFilesWithViolations() throws IOException {
 
         final List<RuleViolation> cachedViolations = reloadedCache.getCachedViolations(sourceFile);
         assertEquals(1, cachedViolations.size(), "Cached rule violations count mismatch");
+        final RuleViolation cachedViolation = cachedViolations.get(0);
+        assertEquals(sourceFile.getDisplayName(), cachedViolation.getFilename());
```

Comment:
I'm not sure the source file has a different display name and path id. I think this should be asserted somewhere

Reply:
That actually depends on the `TextFile` implementation. For `NioTextFile` it is effectively different (a path vs a url, and specially so for zip files), but most other implementations will use the same for both. I don't think asserting that here would be right, as it's an implementation detail of whatever happens within `TextFile.forPath`.